src/components/CodeAndPreview.tsx

```diff
@@ -62,20 +73,38 @@ export function CodeAndPreview({ snippet }: Props) {
     }
   }, [Boolean(snippet)])
 
-  const { toast } = useToast()
-
+  // Update manual edits when snippet changes, with validation
   useEffect(() => {
     if (snippet?.manual_edits_json) {
-      setManualEditsFileContent(snippet.manual_edits_json)
+      try {
+        JSON.parse(snippet.manual_edits_json)
+        setManualEditsFileContent(snippet.manual_edits_json)
```

Comment:
this should be returned as JSON from the api, if it's not we should make it so it does

Reply:
if it's returned as a string, the name should be `manual_edits_json_content`